core/src/main/java/org/jdbi/v3/core/argument/OptionalNestedArgument.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.argument;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * Signals that the annotated element is a optional nested mapped argument type.
+ */
+@Retention(RUNTIME)
+@Target({PARAMETER, FIELD, METHOD})
+public @interface OptionalNestedArgument {
```

Comment:
I think I'd rather we find a way to do this other than a new interface.  I'd prefer an attribute on `@Nested`, or detection of `javax.annotation.Nullable` I think

Reply:
I was thinking `@Nullable` too

Reply:
I like both, I will change this to `@Nullable` and add your crazy idea with :outher.?inner